### PR TITLE
Change devcontainer to use image+features

### PR DESCRIPTION
**Pull request by vzarytovskii**
So we don't have to take care about updates for Debian, etc.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -18,13 +19,20 @@
...
                 "Ionide.Ionide-fsharp",
                 "tintoy.msbuild-project-tools"
             ]
         }
     },
-
-    "onCreateCommand": [ "dotnet", "build", "FSharp.Compiler.Service.sln"],
-    "waitFor": "onCreateCommand"
+    "remoteEnv": {
+        "DOTNET_MULTILEVEL_LOOKUP": "0",
```
Comment (baronfel) [inline .devcontainer/devcontainer.json:34]: This env var isn't required - multilevel lookup was never used on Linux, and in 7.0 we [disabled it by default on Windows as well](https://learn.microsoft.com/en-us/dotnet/core/compatibility/deployment/7.0/multilevel-lookup)

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -18,13 +19,20 @@
         "vscode": {
             // Add the IDs of extensions you want installed when the container is created.
             "extensions": [
-                "ms-dotnettools.csharp",
+                "github.vscode-github-actions",
+                "GitHub.copilot",
+                "GitHub.vscode-github-actions",
```
Comment (baronfel) [inline .devcontainer/devcontainer.json:24]:
```suggestion
                "github.vscode-github-actions",
                "GitHub.copilot",
                "GitHub.vscode-github-actions",
```

Two entries here that differ only on case - does VSCode care?

**Review by baronfel — APPROVED** [review]
I really like the way features make things more composable - always thought the old Dockerfile-based method, while allowing control, was too hard to keep updated!

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (vzarytovskii) [reply .devcontainer/devcontainer.json:24]: Good question. I don't think so?

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (vzarytovskii) [reply .devcontainer/devcontainer.json:34]: I think I copied it over from some example.

**Comment by vzarytovskii** [comment]
Any objections against merging this? @Martin521 i know you use devcontainers a bunch.

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -18,13 +19,18 @@
         "vscode": {
             // Add the IDs of extensions you want installed when the container is created.
             "extensions": [
-                "ms-dotnettools.csharp",
+                "GitHub.copilot",
+                "GitHub.vscode-github-actions",
+                "ms-dotnettools.vscode-dotnet-runtime",
+                "ms-dotnettools.csdevkit",
+                "ms-dotnetools.csharp",
```
Comment (psfinaki) [inline .devcontainer/devcontainer.json:26]: I guess 
```suggestion
                "ms-dotnettools.csharp",
```

**Comment by Martin521** [comment]
Works all fine for me except that the test discovery of the Ionide test explorer fails.
It took me some time to trace it down.
The error message `Couldn't build test projects. Make sure you can build projects with 'dotnet build'` in the "Test Explorer" window was misleading. Looking into the sources, I found that the test discovery first calls `msbuild /t:Build` for the three test projects. Which led me to finding the following build output in the `Ionide msbuild" window:
```
[11:42:39 INFO ] [msbuild] invoking msbuild from /usr/bin/dotnet on /workspaces/fsharp3/tests/FSharp.Compiler.UnitTests/FSharp.Compiler.UnitTests.fsproj for target Build
MSBuild version 17.8.0-preview-23418-03+0125fc9fb for .NET
/usr/share/dotnet/sdk/8.0.100-rc.1.23463.5/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(266,5): error NETSDK1004: Assets file '/workspaces/fsharp3/artifacts/obj/FSharp.Compiler.UnitTests/project.assets.json' not found. Run a NuGet package restore to generate this file. [/workspaces/fsharp3/tests/FSharp.Compiler.UnitTests/FSharp.Compiler.UnitTests.fsproj::TargetFramework=net8.0]
[11:42:41 INFO ] [msbuild] invoking msbuild from /usr/bin/dotnet on /workspaces/fsharp3/tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj for target Build
MSBuild version 17.8.0-preview-23418-03+0125fc9fb for .NET
/usr/share/dotnet/sdk/8.0.100-rc.1.23463.5/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(266,5): error NETSDK1004: Assets file '/workspaces/fsharp3/artifacts/obj/FSharp.Compiler.Service.Tests/project.assets.json' not found. Run a NuGet package restore to generate this file. [/workspaces/fsharp3/tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj::TargetFramework=net8.0]
[11:42:44 INFO ] [msbuild] invoking msbuild from /usr/bin/dotnet on /workspaces/fsharp3/tests/FSharp.Compiler.ComponentTests/FSharp.Compiler.ComponentTests.fsproj for target Build
MSBuild version 17.8.0-preview-23418-03+0125fc9fb for .NET
  FSharp.Core -> /workspaces/fsharp3/artifacts/bin/FSharp.Core/Debug/netstandard2.0/FSharp.Core.dll
  FSharp.Compiler.Interactive.Settings -> /workspaces/fsharp3/artifacts/bin/FSharp.Compiler.Interactive.Settings/Debug/netstandard2.0/FSharp.Compiler.Interactive.Settings.dll
  FSharp.DependencyManager.Nuget -> /workspaces/fsharp3/artifacts/bin/FSharp.DependencyManager.Nuget/Debug/netstandard2.0/FSharp.DependencyManager.Nuget.dll
  FSharp.Compiler.Service -> /workspaces/fsharp3/artifacts/bin/FSharp.Compiler.Service/Debug/netstandard2.0/FSharp.Compiler.Service.dll
  FSharp.Core -> /workspaces/fsharp3/artifacts/bin/FSharp.Core/Debug/netstandard2.1/FSharp.Core.dll
  FSharp.Test.Utilities -> /workspaces/fsharp3/artifacts/bin/FSharp.Test.Utilities/Debug/net8.0/FSharp.Test.Utilities.dll
  FSharp.Compiler.ComponentTests -> /workspaces/fsharp3/artifacts/bin/FSharp.Compiler.ComponentTests/Debug/net8.0/FSharp.Compiler.ComponentTests.dll
```
So msbuild complains about a missing `project.assets.json` for UnitTests and Service.Tests, while ComponentTests build ok.

I found that setting BUILDING_USING_DOTNET=true in devcontainer.json solves the issues, but I don't know if that is the best way to do it.
Or if some setting in one of the msbuild props/proj files can solve it.
Or if Ionide test explorer should just use 'dotnet build' (but I a not sure how soon that can happen).

**Comment by vzarytovskii** [comment]
Yeah, if we set `BUILDING_USING_DOTNET=true` for dev container, then it will be impossible to build using build scripts.

**Comment by Martin521** [comment]
> I don't know if it's possible to set this property to only dotnet commands in the test explorer.

There is no such setting.
Env seems to be the only way, and since VS Code is started from the dev container, it would need to be there.

**Comment by vzarytovskii** [comment]
> > I don't know if it's possible to set this property to only dotnet commands in the test explorer.
> 
> There is no such setting. Env seems to be the only way, and since VS Code is started from the dev container, it would need to be there.

It seems so, yeah. Problem is that in this case, build scripts won't work if someone needs them.

**Comment by Martin521** [comment]
Ok, then I guess you should proceed with this PR.
I will file an issue for Ionide and in the meantime use Directory.build.props.user to set BUILDING_USING_DOTNET.

**Pull request merged**